### PR TITLE
Add delete keys definition for validator keys api

### PR DESCRIPTION
**Pull request by rolfyone**
Added enum type, and some helpers to define authentication responses, rather than re-defining.

Post isn't fully fleshed out, it will need some more groundwork before that will be usable.

partially addresses #4558

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/EndpointMetadata.java
Comment (ajsutton) [inline infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/EndpointMetadata.java]: it feels a little weird to have a static method call here...

File: validator/restapi/src/main/java/tech/pegasys/teku/validator/restapi/ValidatorRestApi.java
Comment (ajsutton) [inline validator/restapi/src/main/java/tech/pegasys/teku/validator/restapi/ValidatorRestApi.java]: This probably shouldn't be here still. :)

File: validator/restapi/src/main/java/tech/pegasys/teku/validator/restapi/apis/schema/DeletionStatus.java
Comment (ajsutton) [inline validator/restapi/src/main/java/tech/pegasys/teku/validator/restapi/apis/schema/DeletionStatus.java:20]: Not a huge fan of breaking the java naming convention here.  Not having to use funny named variables was one of the big advantages of not using the reflection approach.

I wonder if it makes sense to use `DELETED` but override `toString()` to return the external type (`deleted`).  Would need to ensure we use `toString()` instead of `name()` and the deserialiser would have to basically reimplement valueOf.  Could be a follow up PR I guess, I'm just keen to avoid the same downsides we had in the previous approach.

File: infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/types/EnumTypeDefinition.java
Comment (ajsutton) [inline infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/types/EnumTypeDefinition.java:20]: Probably worth adding some tests for this.

**Review by ajsutton — COMMENTED** [review]
Looks good generally.

File: validator/restapi/src/main/java/tech/pegasys/teku/validator/restapi/ValidatorRestApi.java
Comment (rolfyone) [reply validator/restapi/src/main/java/tech/pegasys/teku/validator/restapi/ValidatorRestApi.java]: fair point!

File: infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/EndpointMetadata.java
Comment (ajsutton) [reply infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/EndpointMetadata.java]: Ah meant to come back to this comment.  Given that we didn't wind up testing the `writeOpenApi` method separately, I think I'd make it just be a private method here rather than a static in another class.

File: validator/restapi/src/main/java/tech/pegasys/teku/validator/restapi/apis/schema/DeletionStatus.java
Comment (rolfyone) [reply validator/restapi/src/main/java/tech/pegasys/teku/validator/restapi/apis/schema/DeletionStatus.java:20]: I made a bit of a mess of this change, will be a follow up.

File: infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/types/EnumTypeDefinition.java
Comment (rolfyone) [reply infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/types/EnumTypeDefinition.java:20]: added a serialize test, i think deserialize isn't working currently, will be follow up task.

**Comment by rolfyone** [comment]
I think we can draw a line in this one with the carry-over of:
 - DeletionStatus enumeration to be caps with logic to output lowercase
 - Deserialize tests to flesh out the Deserialization of objects, which will need a builder etc built out most likely

File: infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/types/SerializableObjectTypeDefinition.java
Comment (ajsutton) [inline infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/types/SerializableObjectTypeDefinition.java]: nit: Probably protected if we're aiming for this to be available to subclasses or possibly not needed in this PR.

File: infrastructure/restapi/src/test/java/tech/pegasys/teku/infrastructure/restapi/types/EnumTypeDefinitionTest.java
Comment (ajsutton) [inline infrastructure/restapi/src/test/java/tech/pegasys/teku/infrastructure/restapi/types/EnumTypeDefinitionTest.java]: I think you actually just want this to be 
```suggestion
  DeserializableTypeDefinition<YesNo> definition = DeserializableTypeDefinition.enumOf(YesNo.class);
```

because you only want to test the enum deserialiser, not the object deserialisation wrapper of it as well.

And the test is probably just that it can round trip each value of the enum. `CoreTypesTest` is probably a useful reference.

**Review by ajsutton — APPROVED** [review]
LGTM, but would be quite nice if we can get the `EnumTypeDefinitionTest` sorted before merging - it should be much much simpler if you remove the object wrapper around the enum.

**Pull request merged**
